src/main/java/net/dv8tion/jda/core/entities/MessageChannel.java

```diff
@@ -242,4 +247,38 @@
      *          {@link net.dv8tion.jda.core.Permission#MESSAGE_READ Permission.MESSAGE_READ}
      */
     RestAction<List<Message>> getPinnedMessages();
+
+    default RestAction<Message> editMessageById(String id, String newContent)
+    {
+        return editMessageById(id, new MessageBuilder().appendString(newContent).build());
+    }
+
+    default RestAction<Message> editMessageById(String id, Message newContent)
+    {
+        JSONObject json = new JSONObject().put("content", newContent.getRawContent()).put("tts", newContent.isTTS());
```

Comment:
MessageBuilder does that already. Let the MessageBuilder catch that error and filter back.